precommit.py

```diff
@@ -0,0 +1,82 @@
+#!/usr/bin/env python3
+"""Runs precommit checks on the repository."""
+import argparse
+import os
+import pathlib
+import subprocess
+import sys
+
+
+def main() -> int:
+    """"
+    Execute the main routine.
+    """
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "--overwrite",
+        help=
+        "Overwrites the unformatted source files with the well-formatted code in place. "
+        "If not set, an exception is raised if any of the files do not conform to the style guide.",
+        action='store_true')
+
+    args = parser.parse_args()
+
+    overwrite = bool(args.overwrite)
+
+    repo_root = pathlib.Path(__file__).parent
+    # yapf: disable
+    print("YAPF'ing...")
+    if overwrite:
+        subprocess.check_call([
+            "yapf", "--in-place", "--style=style.yapf", "--recursive", "tests",
+            "lddwrap", "setup.py", "precommit.py", "bin/pylddwrap"
+        ],
+            cwd=repo_root.as_posix())
+    else:
+        subprocess.check_call([
+            "yapf", "--diff", "--style=style.yapf", "--recursive", "tests",
+            "lddwrap", "setup.py", "precommit.py", "bin/pylddwrap"
+        ],
+            cwd=repo_root.as_posix())
+
+    print("Mypy'ing...")
+    subprocess.check_call(["mypy", "lddwrap", "tests", "bin/pylddwrap"],
+                          cwd=repo_root.as_posix())
+
+    print("Pylint'ing...")
+    subprocess.check_call(
+        ["pylint", "--rcfile=pylint.rc", "tests", "lddwrap", "bin/pylddwrap"],
+        cwd=repo_root.as_posix())
+
+    print("Pydocstyle'ing...")
+    subprocess.check_call(["pydocstyle", "lddwrap", "bin/pylddwrap"],
+                          cwd=repo_root.as_posix())
+
+    print("Testing...")
+    env = os.environ.copy()
+    env['ICONTRACT_SLOW'] = 'true'
+
+    subprocess.check_call([
+        "coverage", "run", "--source", "lddwrap", "-m", "unittest", "discover",
+        "tests"
+    ],
+        cwd=repo_root.as_posix())
```

Comment:
could you try to put `cwd=...` on the previous line for readability? (please also see above and below)